### PR TITLE
Update `triangulate_delaunay()` to avoid needless reallocations

### PR DESCRIPTION
**Pull request by Chubercik**
During the review process of the #83353 PR a certain optimization was proposed by one of the members of the Godot team; said optimization can be applied to a related function, which is what this PR accomplishes.

### CODE REVIEW

File: core/math/geometry_2d.h [hunk 1]
```diff
@@ -306,10 +306,12 @@ class Geometry2D {
 		Vector<Delaunay2D::Triangle> tr = Delaunay2D::triangulate(p_points);
 		Vector<int> triangles;
 
+		triangles.resize(3 * tr.size())
+		int32_t *ptr = triangles.ptrw();
```
Comment (AThousandShips) [inline core/math/geometry_2d.h:310]:
```suggestion
		int *ptr = triangles.ptrw();
```

File: core/math/geometry_2d.h [hunk 2]
```diff
@@ -306,10 +306,12 @@ class Geometry2D {
 		Vector<Delaunay2D::Triangle> tr = Delaunay2D::triangulate(p_points);
 		Vector<int> triangles;
 
+		triangles.resize(3 * tr.size())
```
Comment (Chubercik) [inline core/math/geometry_2d.h:309]:
```suggestion
		triangles.resize(3 * tr.size());
```

Missing semicolon 😅

**Review by AThousandShips — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**